contracts/contracts/strategies/MorphoCompoundStrategy.sol

```diff
@@ -0,0 +1,260 @@
+            _pTokens
+        );
+        ORACLE = ICompoundOracle(IMorpho(MORPHO).comptroller().oracle());
+    }
+
+    /**
+     * @dev Approve the spending of all assets by main Morpho contract,
+     *      if for some reason is it necessary.
+     */
+    function safeApproveAllTokens() external override {
+        uint256 assetCount = assetsMapped.length;
+        for (uint256 i = 0; i < assetCount; i++) {
+            address asset = assetsMapped[i];
+
+            // Safe approval
+            IERC20(asset).safeApprove(MORPHO, 0);
+            IERC20(asset).safeApprove(MORPHO, type(uint256).max);
+        }
+    }
+
+    /**
+     * @dev Internal method to respond to the addition of new asset
+     *      We need to approve and allow Morpho to move them
+     * @param _asset Address of the asset to approve
+     * @param _pToken The pToken for the approval
+     */
+    // solhint-disable-next-line no-unused-vars
+    function _abstractSetPToken(address _asset, address _pToken)
+        internal
+        override
+    {
+        IERC20(_asset).safeApprove(MORPHO, 0);
+        IERC20(_asset).safeApprove(MORPHO, type(uint256).max);
+    }
+
+    /**
+     * @dev Collect accumulated rewards and send them to Harvester.
+     */
+    function collectRewardTokens()
+        external
+        override
+        onlyHarvester
+        nonReentrant
+    {
+        /**
+         * Gas considerations. We could query Morpho LENS's `getUserUnclaimedRewards` for each
+         * cToken separately and only claimRewards where it is economically feasible. Each call
+         * (out of 3) costs ~60k gas extra.
+         *
+         * Each extra cToken in the `poolTokens` of `claimRewards` function makes that call
+         * 89-120k more expensive gas wise.
+         *
+         * With Lens query in case where:
+         *  - there is only 1 reward token to collect. Net gas usage in best case would be
+         *    3*60 - 2*120 = -60k -> saving 60k gas
+         *  - there are 2 reward tokens to collect. Net gas usage in best case would be
+         *    3*60 - 120 = 60k -> more expensive for 60k gas
+         *  - there are 3 reward tokens to collect. Net gas usage in best case would be
+         *    3*60 = 180k -> more expensive for 180k gas
+         *
+         * For the above reasoning such "optimization" is not implemented
+         */
+
+        address[] memory poolTokens = new address[](assetsMapped.length);
+        for (uint256 i = 0; i < assetsMapped.length; i++) {
+            poolTokens[i] = assetToPToken[assetsMapped[i]];
+        }
+
+        // solhint-disable-next-line no-unused-vars
+        IMorpho(MORPHO).claimRewards(
+            poolTokens, // The addresses of the underlying protocol's pools to claim rewards from
+            false // Whether to trade the accrued rewards for MORPHO token, with a premium
+        );
+
+        // Transfer COMP to Harvester
+        IERC20 rewardToken = IERC20(rewardTokenAddresses[0]);
```

Comment:
Morpho emits their own rewards? Is this an either/or on rewards, or do we get both?

Reply:
We get the COMP token rewards and can opt to receive rewards in Morpho token. Though I don't see any exchange or AMM that would enable swapping those tokens. So we should stick with receiving COMP protocol rewards. 

There is also a separate [RewardsDistributor](https://etherscan.io/address/0x3B14E5C73e0A56D607A8688098326fD4b4292135) contract that distributes extra rewards  (besides COMP protocol rewards) according to the off chain calculated merkle tree ([see example](https://github.com/morpho-labs/morpho-rewards/blob/main/distribution/age1/epoch1/proofs.json)) These are done once every Age ([their terminology](https://docs.morpho.xyz/usdmorpho/ages-and-epochs) for a 3 month period). 

Since those tokens can not be exchanges right now ([see etherscan token activity](https://etherscan.io/token/0x9994e35db50125e0df82e4c2dde62496ce330999)) I propose we do nothing with them right now and implement some solution down the line there is better support for the token market.

Reply:
Agree.